src/main/java/lotto/domain/Money.java

```diff
@@ -0,0 +1,48 @@
+package lotto.domain;
+
+import java.math.BigDecimal;
+
+public class Money {
+
+    private final BigDecimal amount;
+
+    private Money(BigDecimal amount) {
+        this.amount = amount;
+    }
+
+    public static Money from(int value) {
+        return from(Long.valueOf(value));
+    }
+
+    public static Money from(long value) {
+        verify(value);
+        return new Money(BigDecimal.valueOf(value));
+    }
+
+    public static void verify(long value) {
+        if (BigDecimal.ZERO.compareTo(BigDecimal.valueOf(value)) > 0) {
+            throw new IllegalArgumentException("금액은 음수일 수 없습니다.");
+        }
+    }
+
+    public int divideBy(Money value) {
+        return Integer.parseInt(amount.divide(value.amount).toString());
+    }
```

Comment:
String 으로 변환한 후 다시 int 로 변환하시는 이유가 있으신가요? 😄 
```suggestion
    public int divideBy(Money value) {
        return amount.divide(value.amount).intValue();
    }
```